nip-nostr-connect.md

```diff
@@ -0,0 +1,130 @@
+NIP-Nostr-Connect
+======
+
+Nostr Connect allows Nostr apps to connect with remote signing devices
+------------------------
+
+`draft` `optional` `author:tiero` `author:giowe` `author:vforvalerio87`
+
+## Rationale
+
+Having to enter your Nostr private key on each website or random app _sucks_.
+
+
+## `TL;DR`
+
+
+**App** (*typically a web app, it generates a random ephemeral keypair*) and **Wallet** (*typically a mobile app, it holds the private key of the user that represents his Nostr account*) send to each other `kind:4` encrypted DMs, using a relay of choice. 
```

Comment:
Agreed, but just a thought: would you define the "Saved Messages" of Telegram or "Note to self" of Signal, as being not a legit use case of DMs? In the end, what you doing is to communicating between to yourself accross your own devices (the app running on your browser which generates in YOUR browser an ephemeral key to pair with your other signing app)

Reply:
> ephemeral events could possibly be useful for this.

Main advantage of keeping them persistent is that simplify the work for clients: you can reconstruct the activity log and also give "cloud" sync features out of the box for having multiple devices

Reply:
I don't see the necessity for another kind apart from semantics. One great reason for using kind 4 instead of anything else is that it works right off the bat with any existing relay and requires no work on relay operators' part to work with nostr-connect. Kind 1 would be problematic because those are shown in global feeds and are unencrypted, so using it with specific applications creates clutter for clients and is not secure. Kind 4 has anything that is needed. There's nothing in the NIP that says it can't be used by machine users and is reserved for human users.

Reply:
I think kind 4 could be useful. If I understand this correctly you would see a DM for every new sign in? which could be nice to monitor

Reply:
In the end, I changed my mind to agree. The fact they store DMs is not true. Moreover, if I change the device etc.. and I don't remember all relay URLs of the apps used for my connection, that activity log is lost anyway.

Each signer app should think about how to offer cloud sync/persistence of the list of connected/disconnected apps by other means.

So yes, we can use an ephemeral kind. Any suggestion here?

Reply:
![Screenshot from 2023-01-20 16-53-48](https://user-images.githubusercontent.com/96378/213793514-e77e5004-c26a-443b-8672-500990682b15.png)

Reply:
good idea but the number is wrong LOL, we could use the right number, bhe tnx chatGPT 

https://jalu.ch/coding/primes/#:~:text=24%2C133%20is%20the%20sum%20of%20the%20first%20100%20primes.